src/lambda/commands/uploadLambda.ts

```diff
@@ -43,24 +45,24 @@ interface LambdaFunction {
  * Executes the "Upload Lambda..." command.
  * Allows for uploads of zip files, and both built and unbuilt directories.
  * Does not discriminate on runtime.
- * @param functionNode Function node from AWS Explorer
+ * @param lambdaArg LambdaFunction
+ * @param path Uri to the template.yaml file or the directory the command was invoked from
  */
-export async function uploadLambdaCommand(functionNode: LambdaFunctionNode) {
+export async function uploadLambdaCommand(lambdaArg?: LambdaFunction, path?: vscode.Uri) {
     let result: telemetry.Result = 'Cancelled'
-    const lambda = {
-        name: functionNode.functionName,
-        region: functionNode.regionCode,
-        configuration: functionNode.configuration,
-    }
+    let lambda: LambdaFunction | undefined
 
     try {
-        const response = await new UploadLambdaWizard(lambda).run()
-
-        if (response?.uploadType === 'zip') {
+        const response = await new UploadLambdaWizard(lambdaArg, path).run()
+        if (!response) {
+            return
```

Comment:
this would be something we want to (at least) log. Otherwise "nothing happens" and we don't know why.